datarepo-clienttests/src/main/java/scripts/testscripts/CreateSnapshot.java

```diff
@@ -0,0 +1,199 @@
+import bio.terra.datarepo.model.SnapshotSummaryModel;
+import com.google.cloud.storage.BlobId;
+import common.utils.FileUtils;
+import common.utils.StorageUtils;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.config.TestUserSpecification;
+import scripts.testscripts.baseclasses.SimpleDataset;
+import scripts.utils.DataRepoUtils;
+
+public class CreateSnapshot extends SimpleDataset {
+  private static final Logger logger = LoggerFactory.getLogger(CreateSnapshot.class);
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public CreateSnapshot() {
+    super();
+  }
+
+  private SnapshotModel snapshotModel;
+
+  private static List<BlobId> scratchFiles = new ArrayList<>();
+
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    // create the profile and dataset
+    super.setup(testUsers);
+
+    // get the ApiClient for the snapshot creator, same as the dataset creator
+    ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
+    RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+
+    // load data into the new dataset
+    // note that there's a fileref in the dataset
+    // ingest a file -- TODO CannedTestData.getMeA1KBFile
+    URI sourceUri = new URI("gs://jade-testdata/fileloadprofiletest/1KBfile.txt");
+
+    String loadTag = FileUtils.randomizeName("lookupTest");
+    BulkLoadArrayRequestModel fileLoadModelArray =
+        new BulkLoadArrayRequestModel()
+            .profileId(datasetSummaryModel.getDefaultProfileId())
+            .loadTag(loadTag)
+            .maxFailedFileLoads(0);
+
+    List<String> fileIds = new ArrayList<>();
+    for (int i = 0; i < filesToLoad; i++) {
+      String targetPath = "/testrunner/IngestFile/" + FileUtils.randomizeName("") + ".txt";
+
+      BulkLoadFileModel fileLoadModel =
+          new BulkLoadFileModel()
+              .sourcePath(sourceUri.toString())
+              .description("IngestFile")
+              .mimeType("text/plain")
+              .targetPath(targetPath);
+      fileLoadModelArray.addLoadArrayItem(fileLoadModel);
+      if (fileLoadModelArray.getLoadArray().size() % 1000 == 0 || i == filesToLoad - 1) {
+        JobModel ingestFileJobResponse =
+            repositoryApi.bulkFileLoadArray(datasetSummaryModel.getId(), fileLoadModelArray);
+        ingestFileJobResponse =
+            DataRepoUtils.waitForJobToFinish(repositoryApi, ingestFileJobResponse);
+        BulkLoadArrayResultModel bulkLoadArrayResultModel =
+            DataRepoUtils.expectJobSuccess(
+                repositoryApi, ingestFileJobResponse, BulkLoadArrayResultModel.class);
+
+        try (Stream<BulkLoadFileResultModel> resultStream =
```

Comment:
Yeah, I've been pretty gung-ho about doing this wherever we use streams since my experience has been, if this isn't standard practice, you tend to get unclosed resources that pop up (since you need to constantly rationalize when a stream should or shouldn't be closed....best practice I think is to just always make sure it's closed.

All that being said, I would love for us to use a library like eclipse collections which let's you get rid of 99% of uses of `Stream` instances